### ISSUE TITLE
🐛 BUG: Incognito mode overrides cached data

### ISSUE THREAD

**Issue by aisleo** [dailydotdev/daily#195]
### What went wrong? 🤔

Source and tag list is not saved. I took several minutes to customize my experience on Daily by removing sources I don't like and adding tags I like. But when reloading the new tab page or opening a new tab it resets - adds all sources and removes all tags. Tried this several times but still occurs. I would prefer to use Daily without signing in but I sign up with my GitHub just to check if they are stored but it doesn't work anyway.

### Expected Behavior

Source and tag list should be persistent.

### Actual Behavior

Source and tag list is not persistent.

### Steps to Reproduce Issue

- Remove/Add sources and tags from the "my public sources" list on the left pane
- Reload the page or close and reopen a new tab
- Have a look at the sources list and it is reset as if you have a fresh new install of Daily

#### Operating System

Windows 10

#### Version of Daily

Version 2.20.0

#### Browser

Firefox Developer Edition
Version 82

**Comment by idoshamun** [comment]
Hey @bluffitee,

Thanks for reporting this issue.
I'm very sorry your settings were reset and I can confirm we were able to reproduce this issue.
It's a Firefox-specific bug and I'll do my best to resolve it soon.

**Comment by idoshamun** [comment]
The issue is now fixed and available in version 2.20.1
Please make sure to update the extension and let me know if it works :)

**Comment by bluffitee** [comment]
I noticed the issue still occurs. It seemingly works at first but it resets for some reason. 
After messing with my browser for a while, I think I noticed the issue. I frequently use the browser's private or incognito mode. Do you use the same storage for both regular browsing and incognito (private) browsing mode? If so, that might be the issue. 
- I'm signed in using my GitHub on regular browsing window and made changes to my source list. 
- I opened a new incognito (private) window and opened a new daily tab. 
- And noticed after the incognito window was opened the sources list in my regular window was reset since I'm not logged into my GitHub in incognito mode.
- I'm guessing Daily syncs the same data for both regular and incognito mode, which causes the data in incognito mode to overwrite the data in regular mode.

Until it is fixed, this can be avoided in Firefox by simply disabling Daily in private (incognito) mode.

**Comment by idoshamun** [comment]
Wow! Thanks for the detailed information and for your time.
As far as I know, data cannot be shared across private mode and normal mode.
But it's definitely something to do with the cache. I'll look at it now.

**Comment by idoshamun** [comment]
Investigating further and reading the docs, you're right.
The extension storage is shared between normal and incognito mode.
I guess we should disable saving to cache in incognito.
https://developer.chrome.com/extensions/manifest/incognito

Would you like to contribute a fix to this issue? I will be happy to assist and guide you through...

**Comment by oneminch** [comment]
I can try and have a go at it if @bluffitee doesn't mind 😅

**Comment by idoshamun** [comment]
Let's give @bluffitee some time to answer and if not you can proceed :)

**Comment by bluffitee** [comment]
Have at it! My level of coding knowledge is still low 😄

**Comment by idoshamun** [comment]
@oneminch I guess the issue is yours!
I guess what we need to accomplish is to disable writing to the cache whenever the user browses in incognito mode.
All the writings to the cache happen here:
https://github.com/dailydotdev/daily-apps/blob/master/packages/extension/src/common/cache.js

**Comment by oneminch** [comment]
So I'm guessing I'm going to make a check for whether each browser type is in private mode. Chrome and Firefox are the ones supported, right?

**Comment by idoshamun** [comment]
Exactly. We support also Edge (Chromium-based).
We use Mozilla's polyfill for unified access to the Web Extension API (https://github.com/mozilla/webextension-polyfill).
It's available globally as `browser`.

**Comment by oneminch** [comment]
How can I test the extension? I basically added the code below. I used the `inIncognitoContext` property to check if browser is in private mode or not (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/extension/inIncognitoContext).
```js
if (!browser.extension.inIncognitoContext) {
    return browser.storage.local.set({ [key]: value });
}
```

**Comment by idoshamun** [comment]
Check out the instructions here:
https://github.com/dailydotdev/daily#%EF%B8%8F-setting-up-daily-services
Let me know if you need help

**Comment by oneminch** [comment]
I keep getting this error when I run the Docker setup. I haven't dealt with Docker before I just had a fresh install of it.
```
Traceback (most recent call last):
  File "site-packages\docker\api\client.py", line 205, in _retrieve_server_version
  File "site-packages\docker\api\daemon.py", line 181, in version
  File "site-packages\docker\utils\decorators.py", line 46, in inner
  File "site-packages\docker\api\client.py", line 228, in _get
  File "site-packages\requests\sessions.py", line 543, in get
  File "site-packages\requests\sessions.py", line 530, in request
  File "site-packages\requests\sessions.py", line 643, in send
  File "site-packages\requests\adapters.py", line 449, in send
  File "site-packages\urllib3\connectionpool.py", line 677, in urlopen
  File "site-packages\urllib3\connectionpool.py", line 392, in _make_request
  File "http\client.py", line 1244, in request
  File "http\client.py", line 1290, in _send_request
  File "http\client.py", line 1239, in endheaders
  File "http\client.py", line 1026, in _send_output
  File "http\client.py", line 966, in send
  File "site-packages\docker\transport\npipeconn.py", line 32, in connect
  File "site-packages\docker\transport\npipesocket.py", line 23, in wrapped
  File "site-packages\docker\transport\npipesocket.py", line 72, in connect
  File "site-packages\docker\transport\npipesocket.py", line 59, in connect
pywintypes.error: (2, 'CreateFile', 'The system cannot find the file specified.')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose\cli\main.py", line 67, in main
  File "compose\cli\main.py", line 123, in perform_command
  File "compose\cli\command.py", line 69, in project_from_options
  File "compose\cli\command.py", line 132, in get_project
  File "compose\cli\docker_client.py", line 43, in get_client
  File "compose\cli\docker_client.py", line 170, in docker_client
  File "site-packages\docker\api\client.py", line 188, in __init__
  File "site-packages\docker\api\client.py", line 213, in _retrieve_server_version
docker.errors.DockerException: Error while fetching server API version: (2, 'CreateFile', 'The system cannot find the file specified.')
[15636] Failed to execute script docker-compose
```

**Comment by idoshamun** [comment]
I'm not sure what is this error.
If you want you can submit a PR and I'll do the checks ☺️

**Comment by oneminch** [comment]
I couldn't make Docker work. I have installed it and run the setup commands but it starts getting stuck at some point. So I'll just make changes and submit a PR without testing like before. I'm planning to use this Promise-based script I found on GitHub (https://gist.github.com/jherax/a81c8c132d09cc354a0e2cb911841ff1#file-is-private-mode-js). Should I create a new file for it, export it and import it in cache.js to be used? Also, I'm not sure if I should be posting some credit for the script if I use it. Is that necessary?

Sorry for all the silly questions though. I'm sort of a noob.
